### PR TITLE
Move API into dashboard

### PR DESCRIPTION
**Pull request by SimonTheLeg**
**What this PR does / why we need it**:

Migrates the KKP API code from the https://github.com/kubermatic/kubermatic to here. 

part of #5113 . 

There is still some cleanup to do in both this and the kubermatic repo, but this migrates the main functionality.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind cleanup

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Integrate KKP API into this repository.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by lsviben** [comment]
/approve

Looks like the dashboard e2e tests are failing because of some minio issue.

File: OWNERS_ALIASES [hunk 1]
```diff
@@ -6,3 +6,6 @@ aliases:
     - ahmedwaleedmalik
     - khizerrehan
     - waseem826
+
+  sig-cluster-management: []
```
Comment (ahmedwaleedmalik) [inline OWNERS_ALIASES:10]: Do we really need sig-cluster-management here?

Comment (xrstf) [reply OWNERS_ALIASES:10]: This was just to silence the prow warnings, in actuality we don't need this SIG here, but I didn't want to go through all owners files yet 😁

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve Great job everyone 🍾 
/hold in case some more work is required for this.

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**